### PR TITLE
[ContainerService] Upgrade API version to 2022-09-01 and polish the API

### PR DESCRIPTION
**Pull request by Yao725**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/containerservice/Azure.ResourceManager.ContainerService/tests/autorest.tests.md [hunk 1]
```diff
@@ -1,13 +0,0 @@
-# Generated code configuration
```
Comment (Yao725) [inline sdk/containerservice/Azure.ResourceManager.ContainerService/tests/autorest.tests.md:1]: Temporarily remove the sample generation as the generation will fail.

File: sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs [hunk 2]
```diff
@@ -462,6 +469,14 @@ public partial class AgentPoolUpgradeProfilePropertiesUpgradesItem
         public static bool operator !=(Azure.ResourceManager.ContainerService.Models.AutoScaleExpander left, Azure.ResourceManager.ContainerService.Models.AutoScaleExpander right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class AzureKeyVaultKms
```
Comment (fengzhou-msft) [inline sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs:472]: Add a prefix?

File: sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs [hunk 3]
```diff
@@ -847,6 +865,24 @@ public partial class ContainerServiceUserAssignedIdentity
         public override string ToString() { throw null; }
     }
     [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct KeyVaultNetworkAccessType : System.IEquatable<Azure.ResourceManager.ContainerService.Models.KeyVaultNetworkAccessType>
```
Comment (fengzhou-msft) [inline sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs:868]: Add a prefix?

File: sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs [hunk 4]
```diff
@@ -1054,6 +1091,12 @@ public partial class ManagedClusterNatGatewayProfile
         public int? IdleTimeoutInMinutes { get { throw null; } set { } }
         public int? ManagedOutboundIPCount { get { throw null; } set { } }
     }
+    public partial class ManagedClusterOidcIssuerProfile
+    {
+        public ManagedClusterOidcIssuerProfile() { }
+        public bool? Enabled { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs:1097]: IsEnabled

File: sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs [hunk 5]
```diff
@@ -1054,6 +1091,12 @@ public partial class ManagedClusterNatGatewayProfile
...
+        public string IssuerURL { get { throw null; } }
```
Comment (fengzhou-msft) [inline sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs:1098]: Is the type Uri?

File: sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs [hunk 6]
```diff
@@ -1187,6 +1242,7 @@ public partial class ManagedClusterSku
     public partial class ManagedClusterStorageProfile
     {
         public ManagedClusterStorageProfile() { }
+        public bool? BlobCsiDriverEnabled { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs:1245]: IsBlobCsiDriverEnabled

File: sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs:472]: Should avoid using `Azure` as prefix

Comment (Yao725) [reply sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs]: Renamed to `ManagedClusterSecurityProfileKeyVaultKms`

File: sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs (hunk 3 above)
Comment (Yao725) [reply sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs]: Rename to `ManagedClusterKeyVaultNetworkAccessType`

File: sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs (hunk 4 above)
Comment (Yao725) [reply sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs:1089]: fixed

File: sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs (hunk 5 above)
Comment (Yao725) [reply sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs:1090]: I think it should be, but cannot find samples to prove it, therefore I renamed it to `IssuerUriInfo`

File: sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs (hunk 6 above)
Comment (Yao725) [reply sdk/containerservice/Azure.ResourceManager.ContainerService/api/Azure.ResourceManager.ContainerService.netstandard2.0.cs]: fixed

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**